Fix History Panel bugs and add version mismatch warning in collaborative editor

lmac-1 opened the pull request:
## Description

This PR improves the history panel experience in the collaborative editor by fixing critical bugs that prevented navigation to the "view all history" page and adding a version mismatch warning to prevent user confusion.

Closes #3814 and point 1 of #3940

### History page 500 error when navigating from collaborative editor
<img width="1338" height="644" alt="image" src="https://github.com/user-attachments/assets/6e84521a-9361-4571-a98c-b6abb538e30b" />

- **Frontend**: Fix workflow ID extraction in MiniHistory - was taking "collaborate" instead of the actual UUID from /w/:id/collaborate URLs
- **Backend**: Handle nil case in humanize_workflow when workflow not found
- **Impact**: "View History" button now works from collaborative editor

### Feature: Version mismatch warning banner

Shows a warning when viewing latest workflow but selected run was executed on an older version. Prevents confusion when workflow structure has changed since the run executed. Appears when: 
- Run selected AND viewing "latest" AND versions differ

Example message:

<img width="3020" height="1112" alt="image" src="https://github.com/user-attachments/assets/24fdc0e1-e9ad-4278-990c-14a256c2a490" />

⚠️ This run was executed on version 66, but you're viewing version 67. Steps shown may not match the current workflow structure.

### Testing
- All existing tests passing
- 8 new unit tests for version mismatch detection hook


## Validation steps

**Version mismatch warning:**
You have two choices 
1. Pick an existing workflow where the structure has changed over time in the last 20 runs
2. Make an update to an existing workflow and click Save

Once you have a workflow where the latest version is different to previous version: 
1. Select an old run while viewing "latest" -> Banner appears
2. Switch to matching snapshot version -> Banner disappears
3. Deselect run -> Banner disappears

**Bug fix:**
1. Open workflow in collaborative editor -> Click the little icon next to "View full history" (marked in screenshot above)
2. Verify history page loads AND that it has the correct content (it previously crashed)

## Additional notes for the reviewer

I tried to make the banner look similar to existing banner in LiveView using Tailwind colours.

## AI Usage

Please disclose how you've used AI in this work (it's cool, we just want to know!):

- [x] Code generation (copilot but not intellisense)
- [ ] Learning or fact checking
- [x] Strategy / design
- [ ] Optimisation / refactoring
- [ ] Translation / spellchecking / doc gen
- [ ] Other
- [ ] I have not used AI

You can read more details in our [Responsible AI Policy](https://www.openfn.org/ai#pull-request-templates)

## Pre-submission checklist

- [x] I have performed a **self-review** of my code.
- [ ] I have implemented and tested all related **authorization policies**. (e.g., `:owner`, `:admin`, `:editor`, `:viewer`)
- [ ] I have updated the **changelog**.
- [x] I have ticked a box in "AI usage" in this PR

elias-ba reviewed (approved):
Hey @lmac-1 this is looking great to me from a code point of view. I alkso clicked tested and it looks great. I will leave the UI approvals to Taylor. @taylordowns2000 just wanna flag that the nodes collapsing issue might be out of scope of this work. It has became rare now but it's still happening from time to time in the app on main. I think we may need to have a proper issue for it and do deep diagnostic for it. I do think that it's deeply related to nodes positionning and layout algorithm.

lmac-1 commented:
I think I've fixed the node position bug... but would appreciate some help in understanding how 😅

I was able to consistently reproduce the collapsing nodes issue across multiple branches using these steps:
1. Select a run from the history panel with a "Failed" status.
2. Refresh the page.
3. The nodes will appear collapsed at position (0,0).
4. If you then select a different run and return to the failed run, the nodes display correctly.

I looked into this with Claude, and it suggested the fix I implemented here in this branch in [adc80d9](https://github.com/OpenFn/lightning/pull/3941/commits/adc80d91c6ce039bfa37b8e693689679c3b993ba). The fix works, but I don't fully understand why: The change suppresses the "could not auto-calculate position" warnings when the positions cache is empty on initial render (which is expected before layout runs). But removing these `console.warn()` calls also fixes the visual issue where nodes appear collapsed - not just the console spam. My theory is that the console operations during the render cycle were causing some kind of timing/performance issue that prevented the layout system from completing before React painted the nodes. But I'm not confident in this explanation. @elias-ba - do you have insight into why removing these warnings would fix the visual positioning issue? Is there something about how ReactFlow or the layout system interacts with synchronous operations during render that I'm missing? 

@taylordowns2000 can you retest where you saw the error? It should happen when you have a run selected with a non-successful status and then refresh the page.

lmac-1 commented:
Okay @taylordowns2000, I think I have fixed everything you mentioned:
1. Clicking on runs now automatically updates the version in the header
2. If you then manually select "Latest", the warning appears as expected

I also improved the warning logic so it appears whenever you're viewing a different version than the run's version, not just when viewing "latest". For example:
1. You select a run that was executed on version 4
2. The header automatically switches to v4 (no warning - versions match)
3. You manually change to "Latest" (version 5) → Warning appears: "This run was executed on version 4, but you're viewing version 5..."
4.You manually change to version 3 → Warning still appears: "This run was executed on version 4, but you're viewing version 3..."

This gives you flexibility to compare runs against any workflow version while staying aware of mismatches.
